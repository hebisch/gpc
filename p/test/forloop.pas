program ForLoop;

var
  a: Integer;
  b: Char;
  c: Boolean;
  d: (e, f, g, h, i, j, k);

begin
  for a := 1 to 3 do Write (a);
  WriteLn;
  for a := 1 to 8 do
    begin
      Write (a);
      if a = 4 then Break;
      Write (a)
    end;
  WriteLn;
  for a := 1 to 8 do
    begin
      Write (a);
      if a < 5 then Continue;
      Write (a)
    end;
  WriteLn;
  for a := 3 to 1 do Write (a);
  WriteLn;
  for a := 3 to 2 do Write (a);
  WriteLn;
  for a := 3 to 3 do Write (a);
  WriteLn;
  for a := 3 to 4 do Write (a);
  WriteLn;
  for a := 3 to 5 do Write (a);
  WriteLn;
  for b := 'A' to 'F' do Write (b);
  WriteLn;
  for c := False to True do Write (c);
  WriteLn;
  for d := g to k do Write (Ord (d));
  WriteLn;

  for a := 3 downto 1 do Write (a);
  WriteLn;
  for a := 8 downto 1 do
    begin
      Write (a);
      if a = 4 then Break;
      Write (a)
    end;
  WriteLn;
  for a := 8 downto 1 do
    begin
      Write (a);
      if a < 5 then Continue;
      Write (a)
    end;
  WriteLn;
  for a := 3 downto 1 do Write (a);
  WriteLn;
  for a := 3 downto 2 do Write (a);
  WriteLn;
  for a := 3 downto 3 do Write (a);
  WriteLn;
  for a := 3 downto 4 do Write (a);
  WriteLn;
  for a := 3 downto 5 do Write (a);
  WriteLn;
  for b := 'F' downto 'A' do Write (b);
  WriteLn;
  for c := True downto False do Write (c);
  WriteLn;
  for d := k downto g do Write (Ord (d));
  WriteLn;

  for a in [2, 3, 5, 7] do Write (a);
  WriteLn;
  for a in [2, 3, 5, 7] do
    begin
      Write (a);
      if a = 5 then Break;
      Write (a)
    end;
  WriteLn;
  for a in [2, 3, 5, 7] do
    begin
      Write (a);
      if a < 5 then Continue;
      Write (a)
    end;
  WriteLn;

  for b in ['a', 'c', 'x', 'y'] do Write (b);
  WriteLn;
  for b in ['a', 'c', 'x', 'y'] do
    begin
      Write (b);
      if b = 'x' then Break;
      Write (b)
    end;
  WriteLn;
  for b in ['a', 'c', 'x', 'y'] do
    begin
      Write (b);
      if b < 'x' then Continue;
      Write (b)
    end;
  WriteLn;

  for c in [True, False] do Write (c);
  WriteLn;
  for c in [True, False] do
    begin
      Write (c);
      if not c then Break;
      Write (c)
    end;
  WriteLn;
  for c in [True, False] do
    begin
      Write (c);
      if not c then Continue;
      Write (c)
    end;
  WriteLn;

  for d in [f, h, i, k] do Write (Ord (d));
  WriteLn;
  for d in [f, h, i, k] do
    begin
      Write (Ord (d));
      if d = i then Break;
      Write (Ord (d))
    end;
  WriteLn;
  for d in [f, h, i, k] do
    begin
      Write (Ord (d));
      if d < i then Continue;
      Write (Ord (d))
    end;
  WriteLn;

  {$local W-} for a in [] do {$endlocal} Write (a);
  WriteLn;
  {$local W-} for a in [] do {$endlocal}
    begin
      Write (a);
      if a = 5 then Break;
      Write (a)
    end;
  WriteLn;
  {$local W-} for b in [] do {$endlocal}
    begin
      Write (a);
      if a < 5 then Continue;
      Write (a)
    end;
  WriteLn;

  for a := 1 to 3 do
    for b in ['a', 'c', 'x', 'y'] do
      for c := True downto False do
        for d in [f, h, i, k] do
          Write (a, b, Ord (c), Ord (d), ' ');
  WriteLn ('.')
end.
