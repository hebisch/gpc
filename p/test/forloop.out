123
1122334
123455667788


3
34
345
ABCDEF
FalseTrue
23456
321
887766554
887766554321
321
32
3


FEDCBA
TrueFalse
65432
2357
22335
235577
acxy
aaccx
acxxyy
FalseTrue
False
FalseTrueTrue
1346
11334
134466



1a11 1a13 1a14 1a16 1a01 1a03 1a04 1a06 1c11 1c13 1c14 1c16 1c01 1c03 1c04 1c06 1x11 1x13 1x14 1x16 1x01 1x03 1x04 1x06 1y11 1y13 1y14 1y16 1y01 1y03 1y04 1y06 2a11 2a13 2a14 2a16 2a01 2a03 2a04 2a06 2c11 2c13 2c14 2c16 2c01 2c03 2c04 2c06 2x11 2x13 2x14 2x16 2x01 2x03 2x04 2x06 2y11 2y13 2y14 2y16 2y01 2y03 2y04 2y06 3a11 3a13 3a14 3a16 3a01 3a03 3a04 3a06 3c11 3c13 3c14 3c16 3c01 3c03 3c04 3c06 3x11 3x13 3x14 3x16 3x01 3x03 3x04 3x06 3y11 3y13 3y14 3y16 3y01 3y03 3y04 3y06 .
